operator/apis/http/v1alpha1/httpscaledobject_types.go

```diff
@@ -102,6 +102,9 @@ type HTTPScaledObjectSpec struct {
 	// (optional) Cooldown period value
 	// +optional
 	CooldownPeriod *int32 `json:"scaledownPeriod,omitempty" description:"Cooldown period (seconds) for resources to scale down (Default 300)"`
+	// (optional) Initial period before scaling
+	// +optional
+	InitialCooldownPeriod int32 `json:"initialCooldownPeriod,omitempty" description:"Initial period (seconds) before scaling (Default 0)"`
```

Comment:
```suggestion
	InitialCooldownPeriod *int32 `json:"initialCooldownPeriod,omitempty" description:"Initial period (seconds) before scaling (Default 0)"`
```
I think this is better

Reply:
I can change this, however the scaledObject parameter is not a pointer, which means we'll have to add a nil check.
let me know if you think it's worth.

Reply:
I see, that was an oversight on the upstream KEDA. Let's use pointers here, I will see if I can safely change the upstream as well.

Reply:
yeah, that seems to be correct. In KEDA the [`CooldownPeriod`](
https://github.com/kedacore/keda/blob/27c99dc20a18f85295ef2b32f33534d7545cdd33/apis/keda/v1alpha1/scaledobject_types.go#L91) is a pointer while [`InitialCooldownPeriod`](https://github.com/kedacore/keda/blob/27c99dc20a18f85295ef2b32f33534d7545cdd33/apis/keda/v1alpha1/scaledobject_types.go#L105) is not a pointer. I think it's because default for `CooldownPeriod` is `300` and user setting `0` must be distinguishable from user not setting anything and expecting KEDA to set the default. For `InitialCooldownPeriod`, the default is `0` so setting `nil` would be semantically equivalent.

I'd actually vote for keeping the API consistent with KEDA, `CooldownPeriod` as a pointer and `InitialCooldownPeriod` not a pointer.

Reply:
I'm fine either way, just let me know what to do

Reply:
@wozniakjan I am actually seriously considering changing KEDA api on this. Optional fields should be imho pointers.

@shayrybak will let you know

Reply:
Tracking here: https://github.com/kedacore/keda/issues/6423

Reply:
@zroubalik I see the upstream was merged. should I update this one accordingly? do we want to set default values here or just pass whatever we get to the scaled object?

Reply:
Yes, please also use pointer here and I think that we should keep the defaults to the core, CC @wozniakjan

Reply:
ok, we will probably have to wait for kedacore to cut a release version we can reference with the new type. unless you want to point to a master commit.

Reply:
I've updated the code for now.